pkg/client/interfaces.go

```diff
@@ -81,11 +81,20 @@ type Writer interface {
 // StatusClient knows how to create a client which can update status subresource
 // for kubernetes objects.
 type StatusClient interface {
-	Status() StatusWriter
+	Status() SubResourceWriter
 }
 
-// StatusWriter knows how to update status subresource of a Kubernetes object.
-type StatusWriter interface {
+// SubResourceClient knows how to create a client which can update subresource
+// for kubernetes objects.
+type SubResourceClient interface {
```

Comment:
One thing I am wondering is if this should maybe be called something like `SubResourceCRUDClient` or some such to clarify that this can not deal with other subresources like logs and differentiate it from a potential future non-crud subresource client?

I am not sure though if it is possible to come up with a sensible non-crud subresource client, since that could basically be anything? Maybe that one would just be a standard http client? Ref https://github.com/kubernetes-sigs/controller-runtime/issues/452

Reply:
Thinking about it it again, maybe we should instead just add more verb(s) once we get there?